### PR TITLE
#166816113 user can tag an article 

### PR DESCRIPTION
**Pull request by devPinheiro**
#### What does this PR do?
- It contains a feature for tagging articles by users

#### Description of Task to be completed?
- add Tag model
- create a many-to-many relationship for Tag and Articles Table
- implement tagging article within the createArticle controller

#### How should this be manually tested?
- After cloning the repo, cd into the root folder

- Install all necessary dependencies by running 

  ``npm install``

- Launch the node server with

  ``npm start``

- Register a new user via the ``/api/v1/users/signup`` endpoint

- Create new article via the ``/api/v1/articles`` endpoint

- Using Postman, test the endpoint above with this header:

  ``key: Content-Type``

  ``value: application/json``

- To run local tests using mocha
  
  ``npm run test``

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- PT Story link - `[#166816113] (https://www.pivotaltracker.com/story/show/166816113)`

### CODE REVIEW

File: src/controllers/article.controller.js [hunk 1]
```diff
@@ -289,6 +289,7 @@ export default {
       if (error.status === 404) {
         return Helper.errorResponse(response, error.status, error.message);
       }
+      console.log(error);
```
Comment (fxola) [inline src/controllers/article.controller.js]: Great work Sam. You forgot to take out the console.log statement

Comment (devPinheiro) [reply src/controllers/article.controller.js]: Thanks for pointing that out. I will take it out and would double-check if there are some more.

File: src/db/migrations/20190717164759-create-tag.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('Tags', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      name: {
```
Comment (dharmykoya) [inline src/db/migrations/20190717164759-create-tag.js:10]: Hi @devPinheiro, is there a reason for not validating the name not to allow null cases?

**Comment by dharmykoya** [comment]
Hi @devPinheiro, Good job but did you validate the array in a way numbers won't be accepted?

File: src/db/migrations/20190717164759-create-tag.js (hunk 2 above)
Comment (devPinheiro) [reply src/db/migrations/20190717164759-create-tag.js:10]: Yes, there is a reason for this. The feature is implemented such that users can choose not to add tags to their article. Users can also choose to have any description for their tag.

**Review by tunedev — APPROVED** [review]
Great work @devPinheiro. This LGTM

**Comment by fxola** [comment]
LGTM

**Review by fxola — APPROVED** [review]

**Pull request merged**